jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/EndPoint.java

```diff
@@ -333,6 +333,18 @@ default void write(boolean last, ByteBuffer byteBuffer, Callback callback)
         }
     }
 
+    /**
+     * Cancel any {@link #write(Callback, ByteBuffer...)} operation
+     * in progress.
+     *
+     * @return The callback passed to {@link #write(Callback, ByteBuffer...)} or
+     *         {@code null}
+     */
+    default Callback cancelWrite()
```

Comment:
I don't like having a default implementation here, especially since it does not respect the javadoc'ed contract.